test/git-temp-dir.test.js

```diff
@@ -0,0 +1,71 @@
+import fs from 'fs-extra';
+import path from 'path';
+
+import GitTempDir, {BIN_SCRIPTS} from '../lib/git-temp-dir';
+
+describe('GitTempDir', function() {
+  it('ensures that a temporary directory is populated', async function() {
+    const tempDir = new GitTempDir();
+    await tempDir.ensure();
+
+    const root = tempDir.getRootPath();
+    for (const scriptName in BIN_SCRIPTS) {
+      const script = BIN_SCRIPTS[scriptName];
+      const stat = await fs.stat(path.join(root, script));
+      assert.isTrue(stat.isFile());
+      if (script.endsWith('.sh') && process.platform !== 'win32') {
+        // eslint-disable-next-line no-bitwise
+        assert.isTrue((stat.mode & fs.constants.S_IXUSR) === fs.constants.S_IXUSR);
+      }
+    }
+
+    await tempDir.ensure();
+    assert.strictEqual(root, tempDir.getRootPath());
```

Comment:
just for my own education...

you assign `root = tempDir.getRootPath` on line 11, and then assert that that these are still equal.  With this assertion, are you testing that `ensure()` does not change the root path?

Reply:
> With this assertion, are you testing that `ensure()` does not change the root path?

Yup, exactly. I wanted to assert that multiple calls to `ensure()` wouldn't create new temp directories, basically.